src/Jobs/PullStock.php

```diff
@@ -50,45 +55,65 @@ public function middleware()
         ];
     }
 
-    public function handle() {
+    public function handle()
+    {
         // Resolve Picqer Connection
         $connection = ConnectionFactory::getConnection();
 
         // Filter Exact items based on Daalder product sku
         $code = $this->product->sku;
         $item = new Item($connection);
-        $item = $item->filter("Code eq '".$code."'");
+        $item = $item->filter("Code eq '" . $code . "'");
 
         // Get the Exact Item or return
-        if(count($item) > 0) {
+        if (count($item) > 0) {
             /** @var Item $item */
             $item = $item[0];
         } else {
             $this->fail(
-                'Exact Item not found for Daalder Product with id '. $this->product->id .
+                'Exact Item not found for Daalder Product with id ' . $this->product->id .
                 ' and sku ' . $this->product->sku
             );
         }
 
-        $stockPosition = new StockPosition($connection);
-        $stockPosition = $stockPosition->filter([], '', '', ['itemId' => "guid'{$item->ID}'"]);
+        $itemWarehouse = new ItemWarehouse($connection);
+        $select = "CurrentStock, PlannedStockIn, PlannedStockOut, WarehouseCode";
+        $itemWarehouse = $itemWarehouse->filter("Item eq guid'" . $item->ID . "'", '', $select);
+        $stock = [];
+
 
         // Get the Exact Item or return
-        if(count($stockPosition) > 0) {
-            /** @var StockPosition $stockPosition */
-            $stockPosition = $stockPosition[0];
+        if (count($itemWarehouse) > 0) {
+            /** @var ItemWarehouse $itemWarehouse */
+
+            $stockCollection = collect($itemWarehouse);
```

Comment:
Just to check because I remember the package we use being kinda funky sometimes: will $itemWarehouse never be a singular item, but always a collection? If not, you may want to Collection::wrap here.